src/main/resources/assets/computercraft/lua/rom/programs/delete.lua

```diff
@@ -6,12 +6,20 @@ if args.n < 1 then
 end
 
 for i = 1, args.n do
-    local files = fs.find(shell.resolve(args[i]))
+    local resolvedPath = shell.resolve(args[i])
+    local files = fs.find(resolvedPath)
     if #files > 0 then
         for _, file in ipairs(files) do
-            local ok, err = pcall(fs.delete, file)
-            if not ok then
-                printError((err:gsub("^pcall: ", "")))
+            if file ~= resolvedPath and fs.isReadOnly(file) then
+                print("Skipping read only " .. file)
```

Comment:
Saying "Skipping" seems a little odd, as it seems like a debug message when this is an error. I'd be inclined to say something more direct like `"Cannot delete read-only file /%s"` (and use `printError`).